libcxx/test/std/numerics/numeric.ops/numeric.iota/ranges.iota.pass.cpp

```diff
@@ -0,0 +1,52 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// Testing std::ranges::iota
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17, c++20
+
+#include <cassert>
+#include <numeric>
+#include <array>
+
+#include "test_macros.h"
+#include "test_iterators.h"
+
+// This is pulled directly from the std::iota test
+template <class InIter>
+TEST_CONSTEXPR_CXX20 void test0() {
```

Comment:
Full disclosure: I have no idea what that macro expands to, but ... if this is not supported in C++20, should we be using that here?

Reply:
I was mimicking what llvm does for `std::iota` https://github.com/jamesETsmith/llvm-project/blob/1b7285bf3ee73975fa63acfc5301a9b57a53fca2/libcxx/test/std/numerics/numeric.ops/numeric.iota/iota.pass.cpp#L21-L23

The macro is just a standard specific way to enforce `constexpr`:

https://github.com/jamesETsmith/llvm-project/blob/1b7285bf3ee73975fa63acfc5301a9b57a53fca2/libcxx/test/support/test_macros.h#L166-L170

Since this function has always been `constexpr`, I think I will update it to that explicitly rather than introducing macros.

Reply:
Great!! I think that I should probably start using it in my PRs, too!